sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -0,0 +1,743 @@
+                           covariance_type=cov_type)
+    assert_raise_message(NotFittedError,
+                         "This GaussianMixture instance is not fitted "
+                         "yet. Call 'fit' with appropriate arguments "
+                         "before using this method.", gmm1.score, X)
+
+    # Check score value
+    gmm1.fit(X)
+    gmm_score = gmm1.score(X)
+    gmm_score_proba = gmm1.score_samples(X).mean()
+    assert_almost_equal(gmm_score, gmm_score_proba)
+
+    # Check if the score increase
+    gmm2 = GaussianMixture(n_components=n_components, n_init=1,
+                           max_iter=1000, reg_covar=0, random_state=rng,
+                           covariance_type=cov_type).fit(X)
+    assert_greater(gmm2.score(X), gmm1.score(X))
+
+
+def test_score_samples():
+    random_state = 0
+    cov_type = 'full'
+    rng = np.random.RandomState(random_state)
+    rand_data = RandomData(rng, scale=7)
+    n_components = rand_data.n_components
+    X = rand_data.X[cov_type]
+
+    # Check the error message if we don't call fit
+    gmm = GaussianMixture(n_components=n_components, n_init=1,
+                          max_iter=1, reg_covar=0, random_state=rng,
+                          covariance_type=cov_type)
+    assert_raise_message(NotFittedError,
+                         "This GaussianMixture instance is not fitted "
+                         "yet. Call 'fit' with appropriate arguments "
+                         "before using this method.", gmm.score_samples, X)
+
+    gmm_score_samples = gmm.fit(X).score_samples(X)
+    assert_equal(gmm_score_samples.shape[0], rand_data.n_samples)
+
+
+def test_monotonic_likelihood():
+    random_state = 0
+    rng = np.random.RandomState(random_state)
+    rand_data = RandomData(rng, scale=7)
+    n_components = rand_data.n_components
+
+    for cov_type in COVARIANCE_TYPE:
+        X = rand_data.X[cov_type]
+        gmm = GaussianMixture(n_components=n_components,
+                              covariance_type=cov_type, reg_covar=0,
+                              warm_start=True, max_iter=20)
+        current_log_likelihood = -np.infty
+        for _ in range(100):
+            prev_log_likelihood = current_log_likelihood
+            current_log_likelihood = gmm.fit(X).score(X)
+            assert_greater(current_log_likelihood, prev_log_likelihood)
+
+
+def test_regularisation():
+    random_state = 0
+    rng = np.random.RandomState(random_state)
+    n_samples, n_features = 10, 5
+
+    X = rng.rand(n_samples, n_features)
+    X_diag = X.copy()
+    X_diag[:, 0] = 0
+    X_cov = {'full': X,
+             'tied': X,
+             'diag': X_diag,
+             'spherical': np.zeros((n_samples, n_features))}
+
+    for cov_type in COVARIANCE_TYPE:
+        gmm = GaussianMixture(n_components=n_samples, covariance_type=cov_type,
+                              reg_covar=0)
+        assert_raise_message(ValueError,
```

Comment:
I wonder if we should not introduce a dedicated exception such as `DivergenceError` in `sklearn.exceptions` but maybe we can stick to `ValueError` for now.